src/main/java/ch/njol/skript/expressions/ExprBannerPatterns.java

```diff
@@ -0,0 +1,236 @@
+
+		Consumer<BannerMeta> metaChanger = null;
+		Consumer<Banner> blockChanger = null;
+		switch (mode) {
+			case SET -> {
+				if (placement != null) {
+					metaChanger = bannerMeta -> {
+						if (bannerMeta.numberOfPatterns() < finalPlacement) {
+							int toAdd = finalPlacement - bannerMeta.numberOfPatterns();
+							for (int i = 0; i < toAdd; i++) {
+								bannerMeta.addPattern(new Pattern(DyeColor.GRAY, PatternType.BASE));
+							}
+						}
+						bannerMeta.setPattern(finalPlacement - 1, finalPattern);
+					};
+					blockChanger = banner -> {
+						if (banner.numberOfPatterns() < finalPlacement) {
+							int toAdd = finalPlacement - banner.numberOfPatterns();
+							for (int i = 0; i < toAdd; i++) {
+								banner.addPattern(new Pattern(DyeColor.GRAY, PatternType.BASE));
+							}
+						}
+						banner.setPattern(finalPlacement - 1, finalPattern);
+					};
+				} else {
+					metaChanger = bannerMeta -> {
+						bannerMeta.setPatterns(patternList);
+					};
+					blockChanger = banner -> {
+						banner.setPatterns(patternList);
+					};
+				}
+			}
+			case DELETE -> {
+				if (placement != null) {
+					metaChanger = bannerMeta -> {
+						if (bannerMeta.numberOfPatterns() >= finalPlacement)
+							bannerMeta.removePattern(finalPlacement - 1);
+					};
+					blockChanger = banner -> {
+						if (banner.numberOfPatterns() >= finalPlacement)
+							banner.removePattern(finalPlacement - 1);
+					};
+				} else {
+					metaChanger = bannerMeta -> {
+						bannerMeta.setPatterns(patternList);
+					};
+					blockChanger = banner -> {
+						banner.setPatterns(patternList);
+					};
+				}
+			}
+			case REMOVE -> {
+				metaChanger = bannerMeta -> {
+					List<Pattern> current = bannerMeta.getPatterns();
+					current.removeAll(patternList);
+					bannerMeta.setPatterns(current);
+				};
+				blockChanger = banner -> {
+					List<Pattern> current = banner.getPatterns();
+					current.removeAll(patternList);
+					banner.setPatterns(current);
+				};
+			}
+			case ADD -> {
+				metaChanger = bannerMeta -> {
+					patternList.forEach(bannerMeta::addPattern);
+				};
+				blockChanger = banner -> {
+					patternList.forEach(banner::addPattern);
+				};
+			}
+		}
+
+		for (Object object : objects.getArray(event)) {
+			if (object instanceof Block block && block.getState() instanceof Banner banner) {
+				blockChanger.accept(banner);
+				banner.update(true, false);
+			} else {
+				ItemStack itemStack = ItemUtils.asItemStack(object);
+				if (itemStack == null || !(itemStack.getItemMeta() instanceof BannerMeta bannerMeta))
+					continue;
+				metaChanger.accept(bannerMeta);
+				itemStack.setItemMeta(bannerMeta);
+				if (object instanceof Slot slot) {
+					slot.setItem(itemStack);
+				} else if (object instanceof ItemType itemType) {
+					itemType.setItemMeta(bannerMeta);
+				} else if (object instanceof ItemStack itemStack1) {
+					itemStack1.setItemMeta(bannerMeta);
+				}
+			}
+		}
+
+	}
+
```

Comment:
pretty sure you need to be overriding isSingle, since this isn't a 1:1 relation for the first pattern